openghg/store/base/_base.py

```diff
@@ -162,20 +162,44 @@ def check_hashes(self, filepaths: multiPathType, force: bool) -> Tuple[Dict[str,
 
         return seen, unseen
 
-    def update_metadata(self, data: Dict, additional_metadata: Dict) -> Dict:
+    def add_metakeys(self, force: bool = False) -> dict:
+        """
+        Check metakeys are included from relevant config file and add as the `.metakeys`
+        attributes if not.
+        """
+        if not hasattr(self, "metakeys") or force:
+            try:
+                metakeys = get_metakeys(bucket=self._bucket)[self._data_type]
+            except KeyError:
+                raise ValueError(
+                    f"No metakeys for {self._data_type}, please update metakeys configuration file."
+                )
+
+            self.metakeys = metakeys
+
+        return self.metakeys
+
+    def update_metadata(self, data: Dict, input_parameters: dict, additional_metadata: Dict) -> Dict:
         """This adds additional metadata keys to the metadata within the data dictionary.
 
         Args:
             data: Dictionary containing data and metadata for species
+            input_parameters: Input parameters from read_file.
             additional_metadata: Keys to add to the metadata dictionary
         Returns:
             dict: data dictionary with metadata keys added
         """
+        # Get defined metakeys from the config setup
+        metakeys = self.add_metakeys()
+        required = metakeys["required"]
+
+        input_parameters_required = {key: value for key, value in input_parameters.items() if key in required}
 
         # Basic implemntation of this
         # TODO: Move this somewhere else? This shouldn't need self but does need to understand form of data.
         # TODO: May want to add checks to make sure we're not overwriting anything important.
         for parsed_data in data.values():
+            parsed_data["metadata"].update(input_parameters_required)
```

Comment:
This might be handled in a later PR, but I just want to flag this as something that could _possibly_ change the existing behaviour (if say a parser supplied it's own value for something that was also in `input_parameters_required`)

Reply:
Okay, looking at PR #1113, it looks like the parser metadata gets priority in that case. In Python 3.9, you can do `parsed_data["metadata"] = input_parameters_required | parsed_data["metadata"]` to allow the existing metadata to take preference. But this will happen in PR #1113 anyway, so fine to ignore it here.